app/controllers/admin/notify_downloads_controller.rb

```diff
@@ -0,0 +1,39 @@
+class Admin::NotifyDownloadsController < AdminController
+  before_action :catch_unrecognised_download, only: :show
+
+  def index; end
+
+  def show
+    send_file csv_file, type: 'text/csv', filename: csv_filename
+  end
+
+  private
+
+  def csv_file
+    Tempfile.new.tap do |file|
+      task_notification_list(file).generate
+      file.rewind
+    end
+  end
+
+  def csv_filename
+    "#{params[:id]}_notifications-#{Time.zone.today}.csv"
```

Comment:
There's a part of me which is nervous about having unescaped user content in a filename like this, but I can't actually think what the attack would be.

Reply:
scrap that, I've realised you sanitize it in the before_action